File: leetcode/238.ProductArrayExceptSelf.cc
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> output(nums.size(), 1);
        int runningProd = 1;
        for (int i = 0; i < nums.size(); i++) {
            output[i] *= runningProd;
            runningProd *= nums[i];
        }

        runningProd = 1;
        for (int i = nums.size() - 1; i >= 0; i--) {
            output[i] *= runningProd;
            runningProd *= nums[i];
        }

        return output;
    }
};

File: leetcode/238.ProductArrayExceptSelf.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        runningProd = 1
        res = [1] * len(nums)
        
        # Compute the product of all elements that occur before the current element
        for i in range(0, len(nums)):
            res[i] *= runningProd
            runningProd *= nums[i]
        
        runningProd = 1

        # Compute the product of all elements that occur after the current element
        for i in range(len(nums) - 1, -1, -1):
            res[i] *= runningProd
            runningProd *= nums[i]
        
        return res